src/filtering/markov_kernel.jl

```diff
@@ -87,18 +87,38 @@ function marginalize_cov!(
     C_DxD::AbstractMatrix,
     C_3DxD::AbstractMatrix,
 )
-    D = length(x_curr.μ)
+    _D = size(x_curr.Σ, 1)
     A, b, C = K
     R, M = C_3DxD, C_DxD
 
-    _matmul!(view(R, 1:D, 1:D), x_curr.Σ.R, A')
-    @.. R[D+1:3D, 1:D] = C.R
+    _matmul!(view(R, 1:_D, 1:_D), x_curr.Σ.R, A')
+    @.. R[_D+1:3_D, 1:_D] = C.R
 
     Q_R = triangularize!(R, cachemat=C_DxD)
     copy!(x_out.Σ.R, Q_R)
     return x_out.Σ
 end
 
+function marginalize_cov!(
+    x_out::SRGaussian{T,<:Kronecker.KroneckerProduct},
```

Comment:
This is not just any KroneckerProduct, but specifically those in which the left factor is the identity. So check this!